qdata/ir.py

```diff
@@ -77,31 +94,36 @@ def unpack(args):
 def flatten(*args):
     """Flattens nested lists consisting of Python lists or NestedList objects."""
     obj = unpack(args)
+
     if isinstance(obj, (list, tuple)):
         flat_list = []
         for item in obj:
             flat_list.extend(flatten(item))
         return flat_list
-    elif isinstance(obj, ParsedList):
+
+    if isinstance(obj, ParsedList):
         return flatten(obj.list)
-    else:
-        return [obj]
+
+    return [obj]
 
 
 def format_wires(wire_id_list):
     """Formats wire arguments as strings."""
     if len(wire_id_list) == 1:
         # <id>
         return str(wire_id_list[0])
-    elif len(wire_id_list) == 2:
+
+    if len(wire_id_list) == 2:
         # <id> [<nninteger>]
         return "{}[{}]".format(*wire_id_list)
 
+    raise ValueError(f"Unexpected size of wire list: {len(wire_id_list)}")
+
 
 class ArithmeticOperation:
     """Lightweight class for representing arithmetic operations appearing within the grammar."""
 
-    def __new__(cls, func, *exps):
+    def __new__(cls, func, *exps):  # pylint:disable=unused-argument
```

Comment:
It's a bit confusing to have both `exps` and `args` standing for the same thing. Any preference which one we should keep?

Reply:
Have changed it to args!